hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/SimpleConcurrentFileWritesConflictResolutionStrategy.java

```diff
@@ -68,6 +69,7 @@ public Stream<HoodieInstant> getCandidateInstants(HoodieTableMetaClient metaClie
         .getTimelineOfActions(CollectionUtils.createSet(REPLACE_COMMIT_ACTION, COMPACTION_ACTION))
         .findInstantsAfter(currentInstant.getTimestamp())
         .filterInflightsAndRequested()
+        .filter(i -> (!i.getAction().equals(COMPACTION_ACTION)) || i.getState().equals(REQUESTED))
         .getInstantsAsStream();
```

Comment:
Looks reasonable.

Reply:
@linliu-code We still need to check the compaction for conflict correct?  So instead of filtering out `compaction.inflight`, we should convert `instant.compaction.inflight` to `instant.compaction.request` for checking?  Can you write a test case for this?

Reply:
I guess if the compaction does not really execute before, there is no need to resolve the conflicts, because the log files would slice based on their specific completion time. If there is no confclits for the same file group from multiple writers, then we are good. @linliu-code , we can add some test cases to illustrate this.